### PR TITLE
Update PS Extension docs

### PR DESCRIPTION
**Pull request by sdwheeler**
Fixes AB#4819 - Update PS Extension docs
- Reflowed paragraphs per PowerShell-Docs style guide
- Reorganized content
- Removed outdated references
- Updated main screenshot

### CODE REVIEW

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
This is a great update, I found this clearer and easier to follow!

File: docs/languages/powershell.md [hunk 1]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
 
-## Install the PowerShell extension
+Our test matrix includes the following configurations:
 
-The official PowerShell extension can be installed by following the steps described in the
-[Visual Studio Code User Guide](/docs/editor/extension-marketplace.md) or by going directly to the Visual Studio Code Marketplace and clicking the [Install Button](vscode:extension/ms-vscode.PowerShell).
+- **Windows Server 2022** with Windows PowerShell 5.1 and PowerShell Core 7.2
+- **Windows Server 2019** with Windows PowerShell 5.1 and PowerShell Core 7.2
+- **macOS 11** with PowerShell Core 7.2
+- **Ubuntu 20.04** with PowerShell Core 7.2
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:26]: Fixed

File: docs/languages/powershell.md [hunk 2]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
-![PowerShell extension](images/powershell/PowerShellExtension.png)
+There are two versions of the extension:
 
-## Install from the command line
+- [PowerShell][4] - the fully-tested stable release
+- [PowerShell Preview][5] - a preview release
+  containing new features and other changes, but not fully tested
 
-Alternatively, the PowerShell extension can be installed from any command line (including PowerShell, Cmd, bash) on all platforms using the following command
+Both extensions can be installed, but only one should be enabled. This allows you to choose switch
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:36]: Fixed

File: docs/languages/powershell.md [hunk 3]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
+from the stable to the preview version to try out new features.
 
-```bash
-code --install-extension ms-vscode.powershell
-```
+### Installing from within VS Code
 
-If you are running VS Code [Insiders](/insiders), you will need this command instead:
+The PowerShell extension can be installed Visual Studio Code Marketplace by clicking the
+[Install Button][6]. You can also install the PowerShell extension from within Visual Studio Code by
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:42]: Fixed

File: docs/languages/powershell.md [hunk 4]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
+opening the Extensions view with keyboard shortcut `kb(workbench.view.extensions)` and typing
+"PowerShell" and select the PowerShell extension:
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:44]: Fixed

File: docs/languages/powershell.md [hunk 5]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
+[PSScriptAnalyzer][16] is a PowerShell module that provides a static source code checker for modules
+and scripts. **PSScriptAnalyzer** has rules that verify the quality of PowerShell code. These rules
+are based on PowerShell best practices identified by the PowerShell Team and the community.
+**PSScriptAnalyzer** generates diagnostic records (errors and warnings) to inform users about
+potential code defects and suggests possible solutions for improvements.
 
-The PowerShell extension allows the creation of new Plaster projects by running the **PowerShell: Create New Project from Plaster Template** command from the Command Palette (`kb(workbench.action.showCommands)`).
+The PowerShell extension includes **PSScriptAnalyzer** by default, and automatically performs
+analysis on PowerShell script files being edited in VS Code.
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:125]: Fixed - "you edit"

File: docs/languages/powershell.md [hunk 6]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
 
-![Plaster Project](images/powershell/cpPlasterCommand.png)
+**PSScriptAnalyzer** comes with a collection of built-in rules that check various aspects of
+PowerShell source code such as presence of uninitialized variables, usage of PSCredential Type,
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:128]: Fixed

File: docs/languages/powershell.md [hunk 7]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
-## PSScriptAnalyzer
+To disable **PSScriptAnalyzer**, open your settings, browse Extensions, select the PowerShell
+extension and deselect the checkbox for **PSScriptAnalyzer**.
 
-[PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer#introduction) is a static source code checker for PowerShell modules and scripts. PSScriptAnalyzer checks the quality of PowerShell code by running a set of rules. The rules are based on PowerShell best practices identified by the PowerShell Team and the community. PSScriptAnalyzer generates diagnostic records (errors and warnings) to inform users about potential code defects and suggests possible solutions for improvements.
+![PSScriptAnalyzer Settings][17]
 
-The PowerShell extension includes PSScriptAnalyzer by default, and automatically performs analysis on PowerShell script files being edited in VS Code.
+**PSScriptAnalyzer** also provided code formatting. Automatic document formatting can be invoked
+using the **Format Document** command or the (`kb(editor.action.formatDocument)`) keyboard shortcut.
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:138]: Fixed

File: docs/languages/powershell.md [hunk 8]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
+**Pester** 3.40 pre-installed. To update **Pester** or to install the latest version on other
+platforms follow the [Pester installation instructions][19].
 
-To disable PSScriptAnalyzer, open your settings, browse Extensions, select the PowerShell extension and deselect the checkbox for PSScriptAnalyzer.
+### Plaster integration
 
-![PSScriptAnalyzer Settings](images/powershell/pssaExtensionSetting.png)
+Plaster is a template-based file and project generator written in PowerShell. Its purpose is to
+streamline the creation of PowerShell module projects, Pester tests, DSC configurations and more.
+See the [GitHub Plaster repository][20] for more information, for documentation on Plaster's cmdlets
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:150]: Fixed

File: docs/languages/powershell.md [hunk 9]
```diff
@@ -193,15 +196,83 @@ After installing, add this configuration to your user settings:
...
+- Document folding
+- Document symbols and outline
+- Renaming support
+- Document Formatting
+
+## Example scripts
+
+Example scripts are included with the extension and can be found at the following path.
+
+`C:\Users\<yourusername>\.vscode\extensions\ms-vscode.PowerShell-<version>\examples`
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:216]: Fixed

File: docs/languages/powershell.md [hunk 10]
```diff
@@ -193,15 +196,83 @@ After installing, add this configuration to your user settings:
...
+
+To open or view the examples in Visual Studio Code, run the following from your PowerShell command
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:218]: Fixed

File: docs/languages/powershell.md [hunk 11]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
+Both extensions can be installed, but only one should be enabled. This allows you to choose switch
+from the stable to the preview version to try out new features.
 
-```bash
-code --install-extension ms-vscode.powershell
-```
+### Installing from within VS Code
 
-If you are running VS Code [Insiders](/insiders), you will need this command instead:
+The PowerShell extension can be installed Visual Studio Code Marketplace by clicking the
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:41]: Fixed

File: docs/languages/powershell.md [hunk 12]
```diff
@@ -4,181 +4,184 @@ Area: languages
...
+[Pester][18] is a framework for running unit tests to execute and Windows PowerShell 5.1 comes with
+**Pester** 3.40 pre-installed. To update **Pester** or to install the latest version on other
+platforms follow the [Pester installation instructions][19].
 
-To disable PSScriptAnalyzer, open your settings, browse Extensions, select the PowerShell extension and deselect the checkbox for PSScriptAnalyzer.
+### Plaster integration
 
-![PSScriptAnalyzer Settings](images/powershell/pssaExtensionSetting.png)
+Plaster is a template-based file and project generator written in PowerShell. Its purpose is to
+streamline the creation of PowerShell module projects, Pester tests, DSC configurations and more.
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply docs/languages/powershell.md:149]: Fixed

**Review by michaeltlombardi — APPROVED** [review]

**Comment by gregvanl** [comment]
Looks great! Thanks

**Pull request merged**